### PR TITLE
10 update interface naming and move to plugins directory

### PR DESCRIPTION
**Pull request by lance-wilson**
# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/recenter_tc#10

# Testing Instructions

Run recenter_tc/tests/scripts/abi.tc.Visible.imagery_clean.sh

# Summary

Breaking Changes
================

Update interface naming and move to plugins directory
-------------------------------------------------

*From issue GEOIPS/recenter_tc#10: 2023-04-18*

* Moved interface_modules to plugins/modules

  * Updated related imports
  * Updated entry points
* Renamed filename_format to filename_formatter
* Renamed output_format to output_formatter

::

    moved: recenter_tc/interface_modules to recenter_tc/plugins/modules
    modified: tests/scripts/*
    renamed: recenter_tc/plugins/modules/filename_formats -> recenter_tc/plugins/modules/filename_formatters
    renamed: recenter_tc/plugins/modules/area_def_adjusters -> recenter_tc/plugins/modules/sector_adjusters

# Output

Running tests/scripts/abi.tc.Visible.imagery_clean.sh returns 1, with good image compare, and bad text compare due to GEOIPS_BASEDIR vs GEOIPS_PACKAGES_DIR (probably fixed by NRLMMD-GEOIPS/recenter_tc#9).

### CODE REVIEW

**Comment by mindyls** [comment]
@lance-wilson would you mind reviewing and approving the 1.9.1 PR so we can get it merged first,  before we merge this one ?

Perhaps even merging the 1.9.1 release branch into this branch,  then seeing if the YAML tests pass ?

**Comment by lance-wilson** [comment]
@mindyls I merged the 1.9.1 branch from earlier into this and the tests that I have data for return 0, except that the ascat and metopc windbarb tests seem to be caught halfway between yaml reformatting.  Otherwise I think things are good here.

**Comment by mindyls** [comment]
> @mindyls I merged the 1.9.1 branch from earlier into this and the tests that I have data for return 0, except that the ascat and metopc windbarb tests seem to be caught halfway between yaml reformatting. Otherwise I think things are good here.

@lance-wilson could you remove the ".eggs" directory, and resolve the below conflicts?  Then make sure we're still good?

**Comment by lance-wilson** [comment]
Done and still good.

File: recenter_tc/version.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# file generated by setuptools_scm
```
Comment (mindyls) [inline recenter_tc/version.py:1]: @lance-wilson Could you remove this file as well please?  (It is auto-generatred through setuptools_scm)

Comment (lance-wilson) [reply recenter_tc/version.py]: Done.

**Review by mindyls — CHANGES_REQUESTED** [review]
ModuleNotFoundError: No module named 'geoips.plugins.modules.sector_loaders.trackfile_parsers'

I'm still getting this error - do you have a fix on the geoips repo (in geoips/sector_utils/tc_tracks.py, I believe)?

**Comment by lance-wilson** [comment]
I had to update pyproject.toml for that error.  I also had to update an import in geoips/sector_utils/utils.py.

Is there some place you want that to be merged, since it came from the 165 branch?

I did throw in a try-except block in tc_tracks.py for the windbarb test scripts, but I don't know if we want to keep that or wait until the other yamls are updated.

**Comment by mindyls** [comment]
> I had to update pyproject.toml for that error. I also had to update an import in geoips/sector_utils/utils.py.
> 
> Is there some place you want that to be merged, since it came from the 165 branch?
> 
> I did throw in a try-except block in tc_tracks.py for the windbarb test scripts, but I don't know if we want to keep that or wait until the other yamls are updated.

@lance-wilson I think you can branch off #165 on geoips repo, then checkin those changes and open a PR to 165 (minimal content is fine, just so we can see what changed)

**Comment by lance-wilson** [comment]
This should fix it:  https://github.com/NRLMMD-GEOIPS/geoips/pull/170

**Review by mindyls — APPROVED** [review]
recenter_tc tests in geoips "test_full_install.sh" passed without error.

Package: geoips_full
Total run time: 304 seconds
Number data types run: 11
Number data types failed: 0

**Pull request merged**